src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs

```diff
@@ -211,8 +216,9 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
 
                 var lineView = TextView.GetTextViewLineContainingBufferPosition(point.Value);
 
-                var visualElement = graphicsResult.VisualElement;
-
+                // Looking for IEndOfLineTags and seeing if they exist on the same line as where the
```

Comment:
This code should probably happen before creating the graphics for the inline error adornment.

You also might be able to simplify the check below to if (_endLineTagAggregator.GetTags(lineView.Extent).Any()) continue;

(the GetTags call will only return tags that intersect the span passed in.

Reply:
ah, I could not remember extent. Changed, that logic is much simpler